packages/react-native/React/Views/RCTTVView.m

```diff
@@ -123,18 +114,25 @@ - (void)handleSelect:(UIGestureRecognizer *)r
         self.transform = CGAffineTransformMakeScale(magnification, magnification);
       }
                        completion:^(__unused BOOL finished2) {
-        [self sendSelectNotification:r];
       }];
     }];
-    
-  } else {
-    [self sendSelectNotification:r];
   }
 }
 
-- (void)handleLongSelect:(UIGestureRecognizer *)r
+- (void)handlePress:(UIGestureRecognizer *)r
 {
-    [self sendLongSelectNotification:r];
+  switch (r.state) {
+      case UIGestureRecognizerStateBegan:
+      if (self.onPressIn) self.onPressIn(nil);
+          break;
+      case UIGestureRecognizerStateEnded:
+      case UIGestureRecognizerStateCancelled:
```

Comment:
If the press is cancelled, should we be sending an onPressOut?

Reply:
Maybe that's not something that can happen with TV remote presses...

Reply:
That's a really good question. We probably need to put some more thought into this. I'm not sure it is really possible on TV as you questioned it. Added the clause just to be safe. On the other hand, there's this case where you start the press action and navigate to a different element when the `select` button is held down. So, you start the action when being focused on an element, you end it when you're focused on a completely different one. I think we should treat that case as cancellation but I'm not sure.

That being said, I think we should send onPressOut in the cancelation too. Imagine it like a pointer event, it doesn't mean the button is `pressed`. That's why we probably shouldn't generate an `onPress` or `onLongPress` event in that case. Gotta check the platform standards (incl. web) and see how that case should be handled.

Let's consider that as an edge case and create a bug ticket to refine the implementation later.